src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_SIZE = 16;
+    private static final double RESIZE_BOUND = 0.75;
+    private int size;
+    private int threshold;
+    private Node<K, V>[] nodeArray;
+
+    public MyHashMap() {
+        nodeArray = new Node[DEFAULT_SIZE];
+        threshold = (int) (nodeArray.length * RESIZE_BOUND);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            resize();
+        }
+        Node<K, V> newNode = new Node<>((K) key, (V) value, null);
+        int index = (key == null) ? 0 : key.hashCode() & (nodeArray.length - 1);
+        if (nodeArray[index] == null) {
+            nodeArray[index] = newNode;
+            size++;
+        } else {
+            Node current = nodeArray[index];
+            while (current != null) {
+                if (key == current.key || key != null && key.equals(current.key)) {
```

Comment:
The equality check 'key == current.key' is not safe for comparing keys, especially if the keys are custom objects. It's better to use 'Objects.equals(key, current.key)' for null safety and consistency.